### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by KMironova**
Finished homework_5

### CODE REVIEW

File: src/main/java/com/epam/tc/hw5/pages/UserTablePage.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            verifyDescription(userInformationList.get(i + 2));
+            i += 3;
+        }
+    }
+
+    private void verifyNumberType(String expected) {
+        for (int i = 0; i < userInformationWithNumber.size();) {
+            if (expected.equals(userInformationWithNumber.get(i).getText())) {
+                softAssertions.assertThat(userInformationWithNumber.get(i).isDisplayed());
+                return;
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw5/pages/UserTablePage.java:84]: what are you returning here?

File: src/main/java/com/epam/tc/hw5/pages/UserTablePage.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    private void verifyUser(String expected) {
+        for (WebElement actualUser : usernameList) {
+            if (actualUser.getText().equals(expected)) {
+                softAssertions.assertThat(actualUser.isDisplayed());
+                return;
+            }
+        }
+        Assert.fail("verify User fail");
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw5/pages/UserTablePage.java]: as far as I remember, you have to use only soft assertions.
So, it should be replaced

File: src/test/java/com/epam/tc/hw5/ex1/cucumber/context/TestContext.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.epam.tc.hw5.ex1.cucumber.context;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TestContext {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw5/ex1/cucumber/context/TestContext.java]: you don't need to create separated TestContext classses for each exercise

File: src/test/java/com/epam/tc/hw5/ex1/cucumber/hooks/CucumberHook.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import com.epam.tc.hw5.ex1.cucumber.context.TestContext;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class CucumberHook {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw5/ex1/cucumber/hooks/CucumberHook.java]: you don't have to create separated CucumberHook classes for each exercise.
fix it.

File: src/test/java/com/epam/tc/hw5/ex1/steps/AbstractStep.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.epam.tc.hw5.ex1.steps;
+
+import com.epam.tc.hw5.ex1.cucumber.context.TestContext;
+import com.epam.tc.hw5.pages.DifferentPage;
+import com.epam.tc.hw5.pages.GdiGitHubSite;
+import com.epam.tc.hw5.pages.components.LeftSideMenuComponents;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.WebDriver;
+
+public abstract class AbstractStep {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw5/ex1/steps/AbstractStep.java]: you don't have to create separated AbstractStep classes for each exercise. fix it.

File: src/test/java/com/epam/tc/hw5/ex1/steps/ActionStep.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package com.epam.tc.hw5.ex1.steps;
+
+import io.cucumber.java.en.When;
+
+@SuppressWarnings("checkstyle:Indentation")
+public class ActionStep extends AbstractStep {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw5/ex1/steps/ActionStep.java]: you don't have to create separated ActionStepS classes for each exercise. fix it.

File: src/test/java/com/epam/tc/hw5/ex1/steps/AssertionStep.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package com.epam.tc.hw5.ex1.steps;
+
+import io.cucumber.java.en.Then;
+
+public class AssertionStep extends AbstractStep {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw5/ex1/steps/AssertionStep.java]: you don't have to create separated AssertionStepS classes for each exercise. fix it.

File: src/test/java/com/epam/tc/hw5/ex1/steps/GivenStep.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package com.epam.tc.hw5.ex1.steps;
+
+import com.epam.tc.hw5.ConfProperties;
+import com.epam.tc.hw5.ex1.cucumber.context.TestContext;
+import io.cucumber.java.en.Given;
+
+public class GivenStep extends AbstractStep {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw5/ex1/steps/GivenStep.java]: you don't have to create separated GivenStepS classes for each exercise. fix it.

File: src/test/java/com/epam/tc/hw5/ex1/ExerciseOneTestRun.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package com.epam.tc.hw5.ex1;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions
+public class ExerciseOneTestRun extends AbstractTestNGCucumberTests {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw5/ex1/ExerciseOneTestRun.java]: you don't have to separated test runners for your cucumber tests.
fix it

**Review by ZeroShadows — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/tc/hw5/pages/UserTablePage.java (hunk 1 above)
Comment (KMironova) [reply src/main/java/com/epam/tc/hw5/pages/UserTablePage.java:84]: Here we use 'return' for exit from this method, so we do not return anything.

**Review by ZeroShadows — APPROVED** [review]

**Pull request merged**